tmt/steps/execute/__init__.py

```diff
@@ -245,7 +245,7 @@ def go(self, *args: Any, **kwargs: Any) -> None:
     def discover(self) -> tmt.steps.discover.Discover:
         """ Return discover plugin instance """
         # This is necessary so that upgrade plugin can inject a fake discover
-        return self.step.plan.discover
+        return cast(tmt.steps.discover.Discover, self.step.plan.discover)
```

Comment:
Not sure why, but `tmt/steps/execute/__init__.py:248: error: Returning Any from function declared to return "Discover"` started to appear after I rebased to `main` (after merging #1327). If I remember correctly from annotating `gluetool-modules`, using `cast()` when `@property` decorator used is sometimes necessary due to some bug.

Reply:
You are correct, mymy should be able to infer the correct type. Please, add `cast()` and a big fat TODO item, this might be related to mypy not being allowed to follow imports - once we drop that restriction, `cast()` might become obsolete.

Reply:
Right, added a TODO comment.